wordpress-mobile/WordPress-Android
# UI Modernization Post: Remove post list compact layout

Fixes #19340

This PR is the first in a series which will implement a new look/feel for the Post lists. This focus of this PR was to remove the "compact" layout and any associated logic. No visual changes in this PR.

**Changes Include**
-- Remove compact layout and drawable resources
        - post_list_item_compact
        - post_list_item_skeleton
        - ic_view_post_compact_white_24dp
        - ic_view_post_full_white_24dp
-- Remove `PostListViewLayoutType`
-- Remove tracking layout type changes
-- Remove saving layout type preference
-- Remove `Compact` class
-- Remove compact logic from `PostListMainViewModel`
-- Remove unused dimen resources
-- Update unit tests
-- Search view now uses the base layout
 
**To test:**
- Install and launch the app
- Login
- Navigate from My Site > Posts
- ✅ Verify that the post list works as expected
- Tap the search icon and enter a search query
- ✅ Verify that the search works as expected 

## Regression Notes
1. Potential unintended areas of impact
The app crashes when navigating to posts

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Update unit tests

3. What automated tests I added (or what prevented me from doing so) N/A

PR submission checklist:

UI Changes testing checklist: N/A
